pkg/openwsn/patches/0011-openstack-04-TRAN-add-blocking-sock.patch

```diff
@@ -0,0 +1,549 @@
+index 689a4cb2..0f0d39c2 100644
+--- a/openstack/04-TRAN/sock.c
++++ b/openstack/04-TRAN/sock.c
+@@ -3,8 +3,8 @@
+ #if defined(OPENWSN_UDP_C)
+ 
+ #include "opendefs.h"
+-#include "sock.h"
+-#include "af.h"
++#include "opentimers.h"
++// #include "sock.h"
+ #include "errno.h"
+ #include "packetfunctions.h"
+ #include "openqueue.h"
+@@ -14,12 +14,30 @@
+ #include "scheduler.h"
+ #include "openserial.h"
+ 
+-// ============================ defines ========================================
++#ifdef SOCK_HAS_ASYNC
++#include "net/sock/async.h"
++#endif
++#include "net/sock/udp.h"
++#include "openwsn_log.h"
++#ifdef MODULE_ZTIMER
++#include "ztimer.h"
++#endif /* MODULE_ZTIMER */
++
++//============================ defines ========================================
++
++#ifdef MODULE_OPENWSN
++#define _MSG_TYPE_RECV_PKT (0x1601)
++#ifdef MODULE_ZTIMER
++#define _TIMEOUT_MAGIC     (0xF38A0B63U)
++#define _TIMEOUT_MSG_TYPE  (0x8474)
++#endif /* MODULE_ZTIMER */
++#endif /* MODULE_OPENWSN */
+ 
+-sock_udp_t* udp_socket_list;
++//=========================== variables =======================================
+ 
+-// =========================== variables =======================================
+-// =========================== prototypes ======================================
++static sock_udp_t *_udp_socket_list;
++
++//=========================== prototypes ======================================
+ 
+ static bool _sock_valid_af(uint8_t af);
+ 
+@@ -29,10 +47,14 @@ static void _sock_get_local_addr(open_addr_t* local);
+ 
+ static void _sock_transmit_internal(void);
+ 
++#ifdef MODULE_ZTIMER
++static void _timeout_cb(void *arg);
++#endif
++
+ // ============================= public ========================================
+ 
+ void sock_udp_init(void) {
+-    udp_socket_list = NULL;
++    _udp_socket_list = NULL;
+ }
+ 
+ int sock_udp_create(sock_udp_t* sock, const sock_udp_ep_t* local, const sock_udp_ep_t* remote, uint16_t flags) {
+@@ -42,10 +64,14 @@ int sock_udp_create(sock_udp_t* sock, const sock_udp_ep_t* local, const sock_udp
+         return -EINVAL;
+     }
+ 
++#ifdef MODULE_OPENWSN
++    mbox_init(&sock->mbox, sock->mbox_queue, OPENWSN_SOCK_MBOX_SIZE);
++#endif
++
+     memset(&sock->gen_sock.local, 0, sizeof(sock_udp_ep_t));
+ 
+     if (local != NULL) {
+-        current = udp_socket_list;
++        current = _udp_socket_list;
+ 
+         while (current != NULL) {
+             if (current->gen_sock.local.port == local->port) {
+@@ -73,10 +99,13 @@ int sock_udp_create(sock_udp_t* sock, const sock_udp_ep_t* local, const sock_udp
+     }
+ 
+     sock->gen_sock.flags = flags;
++
++#ifdef SOCK_HAS_ASYNC
+     sock->async_cb = NULL;
++#endif
+ 
+-    sock->next = udp_socket_list;
+-    udp_socket_list = sock;
++    sock->next = _udp_socket_list;
++    _udp_socket_list = sock;
+ 
+     return 0;
+ }
+@@ -144,21 +173,21 @@ int sock_udp_send(sock_udp_t* sock, const void* data, size_t len, const sock_udp
+ 
+     memcpy(pkt->payload, data, len);
```

Comment:
Same here, new signature is:

```C
ssize_t sock_udp_send_aux(sock_udp_t *sock, const void *data, size_t len,
                          const sock_udp_ep_t *remote, sock_udp_aux_tx_t *aux);
```

Again, the `aux` parameter can be ignored and features be added as follow up.

Reply:
See https://github.com/RIOT-OS/RIOT/pull/15536#issuecomment-737902817